openapi.yml

```diff
@@ -32,71 +55,85 @@ paths:
-      description: Nonce of the batch in which the order is valid. uint32.
-      type: integer
-    OrderNew:
+    Tip:
+      description: The tip that is used to pay solvers. uint256.
+      type: object
+      properties:
+        expirationDate:
+          description: |
+            Expiration date of the offered tip. Order service might not accept 
+            the tip after this expiration date. Encoded as unix timestamp.
+          type: integer
+        tipAmount:
+          description: Amount of tip in specified sellToken
+          $ref: "#/components/schemas/TokenAmount"
+    OrderType:
+      description: Is this a buy order or sell order?
+      type: string
+      enum: [buy, sell]
+    OrderCreation:
       description: Data a user provides when creating a new order.
       type: object
       properties:
+        sellToken:
+          description: "ERC20 token to be sold"
+          $ref: "#/components/schemas/Address"
         buyToken:
+          description: "ERC20 token to be bought"
           $ref: "#/components/schemas/Address"
-        buyAmount:
-          $ref: "#/components/schemas/TokenAmount"
         sellAmount:
+          description: "Amount of sellToken to be sold in atoms"
           $ref: "#/components/schemas/TokenAmount"
-        sellToken:
-          $ref: "#/components/schemas/Address"
-        sellTokenTip:
+        buyAmount:
+          description: "Amount of buyToken to be bought in atoms"
           $ref: "#/components/schemas/TokenAmount"
-        orderType:
-          description: Is this a buy order or sell order?
-          type: string
-          enum: [buy, sell]
-        fillType:
-          description: Is this a fill-or-kill order a partially fillable order?
-          type: string
-          enum: [fillorkill, partial]
-        nonce:
-          $ref: "#/components/schemas/Nonce"
         validTo:
-          description: Time offset in seconds from epoch until which the order is valid. uint32.
+          description: Unix timestamp until the order is valid. uint32.
+          type: integer
+        appData:
+          description: |
+            Arbitrary identifier sent along with the order.
+            Could be used to track the interface or other meta-aspects of the order. uint32 encoded
           type: integer
+        tip:
+          description: "Tip to compensate solvers in atoms"
+          $ref: "#/components/schemas/TokenAmount"
+        orderType:
+          description: "The type is either a buy or sell order"
+          $ref: "#/components/schemas/OrderType"
+        partiallyFillable:
+          description: Is this a fill-or-kill order or a partially fillable order?
+          type: boolean
         signature:
           description: 65 bytes encoded as hex without `0x` prefix. v + r + s from the spec.
           example: "0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
     OrderExtra:
```

Comment:
Can `OrderExtra` also include `UUID`, so that we get complete orders it when querying `/api/v1/orders`